spec/components/trainees/confirmation/diversity/view_preview.rb

```diff
@@ -6,43 +6,43 @@ module Confirmation
     module Diversity
       class ViewPreview < ViewComponent::Preview
         def not_disclosed
-          render(Trainees::Confirmation::Diversity::View.new(trainee: mock_trainee_with_no_disclosure))
+          render(Trainees::Confirmation::Diversity::View.new(data_model: mock_trainee_with_no_disclosure))
         end
 
         def disclosed_with_no_ethnic_group
-          render(Trainees::Confirmation::Diversity::View.new(trainee: mock_trainee_with_disclosure_and_no_ethnic_group))
+          render(Trainees::Confirmation::Diversity::View.new(data_model: mock_trainee_with_disclosure_and_no_ethnic_group))
         end
 
         def disclosed_with_ethnic_group
-          render(Trainees::Confirmation::Diversity::View.new(trainee: mock_trainee_with_disclosure_and_ethnic_group))
+          render(Trainees::Confirmation::Diversity::View.new(data_model: mock_trainee_with_disclosure_and_ethnic_group))
         end
 
         def disclosed_with_no_disabilities
-          render(Trainees::Confirmation::Diversity::View.new(trainee: mock_trainee_with_disclosure_and_no_disabilities))
+          render(Trainees::Confirmation::Diversity::View.new(data_model: mock_trainee_with_disclosure_and_no_disabilities))
         end
 
         def disclosed_with_disabilities
-          render(Trainees::Confirmation::Diversity::View.new(trainee: mock_trainee_with_disclosure_and_disabilities))
+          render(Trainees::Confirmation::Diversity::View.new(data_model: mock_trainee_with_disclosure_and_disabilities))
         end
 
       private
 
         def mock_trainee_with_no_disclosure
-          OpenStruct.new({
+          Trainee.new({
```

Comment:
We have some files using `OpenStruct` and some referencing actual models. We should decide between the two to be consistent. These components shouldn't care what model you pass in as long as you just provide the right properties. Personally, i'm not sure if using the actual model makes sense as it's loaded with various other properties the component may not care about. Maybe something to discuss with the rest of the team?

Reply:
FYI this doesn't need to be addressed in this PR

Reply:
I had to change it to `Trainee` because the component `Trainees::Confirmation::Diversity::View` needs to implement the following method:

```ruby
def trainee
  data_model.is_a?(Trainee) ? data_model : data_model.trainee
end
```

Happy to discuss the wider implications of this with the team.